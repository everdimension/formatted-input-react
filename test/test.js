import React from 'react';
import {
  Simulate,
  renderIntoDocument,
  findRenderedDOMComponentWithTag,
} from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import NumberInput from '../src';

describe('Rendering', () => {
  it('Renders a component of type "input"', () => {
    const component = renderer.create(
      <NumberInput
        name="test-input"
        value=""
      />
    );
    const tree = component.toJSON();
    expect(tree.type).toBe('input');
  });

  it('Renders a component with correct props', () => {
    const component = renderer.create(
      <NumberInput
        name="test-input"
        value=""
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.props).toMatchSnapshot();
  });
});

describe('Behavior', () => {
  class TestingParentControlled extends React.Component {
    constructor() {
      super();
      this.state = { value: '' };
    }

    render() {
      return (
        <NumberInput
          value={this.state.value}
          onChange={() => {}}
        />
      );
    }
  }

  class TestingParentUncontrolled extends React.Component {
    constructor() {
      super();
      this.state = { value: '' };
    }

    render() {
      return (
        <NumberInput
          defaultValue={this.state.value}
          onChange={() => {}}
        />
      );
    }
  }

  describe('Passes updated value via onChange callback', () => {
    const recorder = {};
    const inputName = 'test-input';
    const expectedValue = '23';

    const root = renderIntoDocument(
      <NumberInput
        name={inputName}
        value=""
        onChange={(name, value) => (recorder[name] = value)}
      />
    );

    const input = findRenderedDOMComponentWithTag(root, 'input');
    input.value = String(expectedValue);
    Simulate.change(input, {
      target: input,
    });

    it('passes name as the first argument value', () => {
      expect(inputName in recorder).toBe(true);
    });

    it('passes correct value as the second argument', () => {
      expect(typeof recorder[inputName]).toBe('string');
      expect(recorder[inputName]).toBe(expectedValue);
    });
  });

  describe('Controlled input', () => {
    it('Updates when new props are received', () => {
      const root = renderIntoDocument(
        <TestingParentControlled />
      );
      const input = findRenderedDOMComponentWithTag(root, 'input');

      /* component was not yet updated */
      expect(root.state.value).toBe('');
      expect(input.value).toBe('');

      /* pass new props to NumberInput */
      root.setState({ value: '123' });
      expect(input.value).toBe('123');
    });
  });

  describe('Uncontrolled NumberInput', () => {
    it('Doesn\'t update when new props are received', () => {
      const root = renderIntoDocument(
        <TestingParentUncontrolled />
      );
      const input = findRenderedDOMComponentWithTag(root, 'input');

      /* component was not yet updated */
      expect(root.state.value).toBe('');
      expect(input.value).toBe('');

      /* pass new props to NumberInput */
      root.setState({ value: '123' });
      expect(input.value).toBe('');
    });
  });
});
