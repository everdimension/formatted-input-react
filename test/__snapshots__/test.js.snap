// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders a component with correct props 1`] = `
<input
  name="test-input"
  onChange={[Function]}
  type="text"
  value=""
/>
`;

exports[`Rendering Renders a component with correct props 2`] = `
Object {
  "name": "test-input",
  "onChange": [Function],
  "type": "text",
  "value": "",
}
`;
